api/api.py

```diff
@@ -0,0 +1,659 @@
+            },
+        }
+    )
+
+
+@app.get("/courts/")
+def read_courts(
+    response: Response,
+    request: Request,
+    limit: Optional[int] = Query(
+        100, description="Limit the number of results, -1 for all, default 100 results"
+    ),
+    search: Optional[str] = Query(None, description="Court name to filter by"),
+    db: Session = Depends(get_db),
+):
+    """API endpoint to get court types with optional search and limit parameters"""
+
+    params = request.query_params
+    query_param_list = ["search", "limit"]
+    validate_query_params(params, query_param_list)
+
+    query = db.query(Court)
+    if search is not None:
+        query = query.where(Court.court_name.ilike(f"%{search}%"))
+
+    if limit != -1:
+        query = query.limit(limit)
+
+    result = query.all()
+    if not result:
+        return no_matches(response, "court names")
+    return result
+
+
+@app.get("/judges/", response_model=List[JudgeModel])
+def read_judges(
+    request: Request,
+    response: Response,
+    limit: Optional[int] = Query(
+        100, description="Limit the number of results, -1 for all, default 100 results"
+    ),
+    search: Optional[str] = Query(None, description="Judge name to filter by"),
+    db: Session = Depends(get_db),
+):
+    """API endpoint to get judge names with optional search and limit parameters"""
+    params = request.query_params
+    query_param_list = ["search", "limit"]
+    validate_query_params(params, query_param_list)
+
+    query = db.query(Judge)
+    if search is not None:
+        query = query.where(Judge.judge_name.ilike(f"%{search}%"))
+
+    if limit != -1:
+        query = query.limit(limit)
+
+    result = query.all()
+    if not result:
+        return no_matches(response, "judge names")
+
+    return result
+
+
+@app.get("/lawyers/", response_model=List[LawyerModel])
+def read_lawyers(
+    request: Request,
+    response: Response,
+    limit: Optional[int] = Query(
+        100, description="Limit the number of results, -1 for all, default 100 results"
+    ),
+    lawyer: Optional[str] = Query(None, description="Lawyer name to filter by"),
+    law_firm: Optional[str] = Query(None, description="Law firm name to filter by"),
+    db: Session = Depends(get_db),
+):
+    """API endpoint to get lawyer and law firm names with optional search and limit parameters"""
+    params = request.query_params
+    query_param_list = ["lawyer", "law_firm", "limit"]
+    validate_query_params(params, query_param_list)
+
+    query = db.query(Lawyer).options(selectinload(Lawyer.law_firm))
+
+    if lawyer is not None:
+        query = query.where(Lawyer.lawyer_name.ilike(f"%{lawyer}%"))
+
+    if law_firm is not None:
+        query = query.join(Lawyer.law_firm).where(
+            LawFirm.law_firm_name.ilike(f"%{law_firm}%")
+        )
+
+    if limit != -1:
+        query = query.limit(limit)
+
+    result = query.all()
+    if not result:
+        return no_matches(response, "lawyer names")
```

Comment:
A little out of scope but another way to do this is by creating a  function to do all this as these lines are repeated in all the functions however, I understand due to all the interfacing tables and different dependencies you would have to pass into the function.